src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1482,6 +1482,45 @@ def update_abbr_record(model: Model, rv_trans):
     return model, trans
 
 
+def add_efim_records(control_stream, idx_cov, last_est_rec):
+    data_record = str(control_stream.get_records("DATA")[0]).rstrip()
+    input_record = str(control_stream.get_records("INPUT")[0]).rstrip()
+    efim_records = [
+        "$PROBLEM DESIGN",
+        f"{data_record} REWIND",
+        f"{input_record}",
+        "$MSFI efim.msf",
+        "$DESIGN APPROX=FO FIMDIAG=1 GROUPSIZE=1 OFVTYPE=1",
+    ]
+
+    # EFIM records need to be added in reverse order to the control stream
+    new_records = [create_record(f"{rec}\n") for rec in efim_records]
+    for new_record in reversed(new_records):
+        control_stream = control_stream.insert_record(new_record, at_index=idx_cov + 1)
+
+    # MSFO in $ESTIMATION needed for EFIM in next $PROBLEM
```

Comment:
I think this comment is perhaps not necessary, from the code we can see that MSFO is added and we know it's needed since we're adding it.